### PR TITLE
Actions form refactor

### PR DESCRIPTION
**Pull request by TomaszChmielPagepro**
https://tw.auroracreation.com/app/tasks/15582197

### CODE REVIEW

File: src/components/v5/common/ActionSidebar/partials/PopularActions.tsx [hunk 1]
```diff
@@ -2,11 +2,34 @@ import React, { FC } from 'react';
 import { useIntl } from 'react-intl';
 
 import { TextButton } from '~v5/shared/Button';
-import { Actions } from '~constants/actions';
+import { ACTION } from '~constants/actions';
 import { PopularActionsProps } from '../types';
 
 const displayName = 'v5.common.ActionSidebar.partials.PopularActions';
 
+const POPULAR_ACTIONS = [
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionSidebar/partials/PopularActions.tsx:10]: Type.
Please also moveit to consts

File: src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  ] = useToggle();
+  const selectedOption = teamsOptions.options.find(
+    (option) => option.value === selectedTeam,
+  );
+
+  return (
+    <div className="sm:relative w-full">
+      <button
+        type="button"
+        className={clsx('flex text-md transition-colors hover:text-blue-400', {
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx:41]: Hover on desktop only

File: src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            }
+          />
+        ) : (
+          formatMessage({ id: 'actionSidebar.selectTeam' })
+        )}
+      </button>
+      {isTeamSelectVisible && (
+        <SearchSelect
+          ref={registerContainerRef}
+          items={[teamsOptions]}
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx:62]: Isn't it already an array?

File: src/components/v5/common/ActionSidebar/partials/forms/CreateNewTeamForm/CreateNewTeamForm.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import React, { FC } from 'react';
+
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionSidebar/partials/forms/CreateNewTeamForm/CreateNewTeamForm.tsx:2]: Redundant empty line

File: src/components/v5/common/ActionSidebar/types.ts [hunk 5]
```diff
@@ -13,3 +16,20 @@ export type ErrorBannerProps = {
   title: MessageDescriptor | string;
   actionText?: MessageDescriptor | string;
 };
+
+//
+export type ActionFormOptions = UseFormProps & {
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionSidebar/types.ts:21]: Can you use interface here? Also, what is `//` in the line above for?

File: src/components/v5/common/SubNavigation/partials/DropdownContent/DecideDropdown.tsx [hunk 6]
```diff
@@ -9,14 +9,16 @@ import styles from './DropdownContent.module.css';
 import LearnMore from '~shared/Extensions/LearnMore';
 import TitleLabel from '~v5/shared/TitleLabel';
 import { useActionSidebarContext } from '~context/ActionSidebarContext';
-import { Actions } from '~constants/actions';
+// import { Actions } from '~constants/actions';
```
Comment (joanna-pagepro) [inline src/components/v5/common/SubNavigation/partials/DropdownContent/DecideDropdown.tsx:12]: :)

File: src/components/v5/common/SubNavigation/partials/DropdownContent/DecideDropdown.tsx [hunk 7]
```diff
@@ -29,16 +31,16 @@ const DecideDropdown: FC<PropsWithChildren> = () => {
           title={MSG.createDecision}
           description={MSG.createDecisionDescription}
           onClick={() => {
-            setSelectedAction(Actions.CREATE_DECISION);
-            toggleActionBar();
+            // setSelectedAction(Actions.CREATE_DECISION);
```
Comment (joanna-pagepro) [inline src/components/v5/common/SubNavigation/partials/DropdownContent/DecideDropdown.tsx:34]: :)

File: src/components/v5/common/SubNavigation/partials/DropdownContent/ManageDropdown.tsx [hunk 8]
```diff
@@ -29,32 +30,32 @@ const ManageDropdown: FC<PropsWithChildren> = () => {
           title={MSG.manageTeams}
           description={MSG.manageTeamsDescription}
           onClick={() => {
-            setSelectedAction(Actions.EDIT_EXISTING_TEAM);
-            toggleActionBar();
+            // setSelectedAction(Actions.EDIT_EXISTING_TEAM);
```
Comment (joanna-pagepro) [inline src/components/v5/common/SubNavigation/partials/DropdownContent/ManageDropdown.tsx:33]: :)

File: src/components/v5/common/SubNavigation/partials/DropdownContent/PayDropdown.tsx [hunk 9]
```diff
@@ -30,33 +31,33 @@ const PayDropdown: FC<PropsWithChildren> = () => {
           title={MSG.singlePayments}
           description={MSG.singlePaymentsDescription}
           onClick={() => {
-            setSelectedAction(Actions.SIMPLE_PAYMENT);
-            toggleActionBar();
+            // setSelectedAction(Actions.SIMPLE_PAYMENT);
```
Comment (joanna-pagepro) [inline src/components/v5/common/SubNavigation/partials/DropdownContent/PayDropdown.tsx:34]: :)

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

File: src/components/v5/common/ActionFormRow/hooks.ts [hunk 10]
```diff
@@ -1,9 +1,13 @@
-import { useActionSidebarContext } from '~context/ActionSidebarContext';
+// import { useActionSidebarContext } from '~context/ActionSidebarContext';
+import { useWatch } from 'react-hook-form';
 import { ActionSidebarRowFieldNameEnum } from './enums';
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionFormRow/hooks.ts]: I will fix orders in another pr by adding a rule to eslint

File: src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx (hunk 3 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx:62]: No idea, I don't want to break this now :D

File: src/components/v5/common/ActionFormRow/hooks.ts [hunk 11]
```diff
@@ -1,9 +1,13 @@
-import { useActionSidebarContext } from '~context/ActionSidebarContext';
+// import { useActionSidebarContext } from '~context/ActionSidebarContext';
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionFormRow/hooks.ts]: Done

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx [hunk 12]
```diff
@@ -1,148 +1,145 @@
...
-      <div className="px-6 py-8 overflow-scroll h-[40rem]">
-        {!selectedAction && (
-          <>
+  return (
+    <div
+      className={clsx(
+        `fixed top-0 right-0 bottom-0 w-full h-full bg-base-white rounded-bl-lg border-l border-gray-200 shadow-default transition-all z-[60] flex flex-col`,
+        {
+          'sm:max-w-[43.375rem]': !isSidebarFullscreen,
+          'max-w-full': isSidebarFullscreen,
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/ActionSidebar.tsx]: Done

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx [hunk 13]
```diff
@@ -1,148 +1,145 @@
...
 
-  const formContent = (
-    <>
-      <div className="px-6 py-8 overflow-scroll h-[40rem]">
-        {!selectedAction && (
-          <>
+  return (
+    <div
+      className={clsx(
+        `fixed top-0 right-0 bottom-0 w-full h-full bg-base-white rounded-bl-lg border-l border-gray-200 shadow-default transition-all z-[60] flex flex-col`,
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/ActionSidebar.tsx]: Done

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx [hunk 14]
```diff
@@ -1,148 +1,145 @@
...
-                actionsWithErrorBanners || isFieldError ? 'error' : 'warning'
-              }
-              title={<FormattedMessage id={prepareNofiticationTitle()} />}
-              action={{
-                type: 'call-to-action',
-                actionText: actionsWithErrorBanners ? (
-                  <FormattedMessage id="learn.more" />
-                ) : null,
-              }}
+            {FormComponent && <FormComponent {...{ getFormOptions }} />}
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/ActionSidebar.tsx]: Done

File: src/components/v5/common/ActionSidebar/ActionTypeSelect.tsx [hunk 15]
```diff
@@ -1,65 +1,60 @@
...
     <ActionSidebarRow
-      fieldName={ActionSidebarRowFieldNameEnum.ACTION_TYPE}
+      fieldName={ACTION_TYPE_FIELD_NAME}
       iconName="file-plus"
-      title={{ id: 'actionSidebar.actionType' }}
+      title={intl.formatMessage({ id: 'actionSidebar.actionType' })}
     >
       <>
-        {!selectedAction && (
+        {!actionType && (
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/ActionTypeSelect.tsx]: Done

File: src/components/v5/common/ActionSidebar/partials/ColonyDetailsFields/ColonyDetailsFields.tsx [hunk 16]
```diff
@@ -60,14 +56,14 @@ const ColonyDetailsFields = () => {
...
-        isError={formErrors?.colonyDescription}
-        fieldName={ActionSidebarRowFieldNameEnum.COLONY_DESCRIPTION}
+        title={<FormattedMessage id="actionSidebar.colonyDescription" />}
+        fieldName="colonyDescription"
       >
         <DefaultField
           name="colonyDescription"
-          placeholder={{ id: 'actionSidebar.colonyDescription.placeholder' }}
-          isError={formErrors?.colonyDescription}
+          placeholder={intl.formatMessage({
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/ColonyDetailsFields/ColonyDetailsFields.tsx:63]: Done

File: src/components/v5/common/ActionSidebar/partials/ColonyDetailsFields/ColonyDetailsFields.tsx [hunk 17]
```diff
@@ -1,45 +1,41 @@
...
-import { useActionFormContext } from '~v5/common/ActionSidebar/partials/ActionForm/ActionFormContext';
 import { useColonyContext } from '~hooks';
 import ColonyAvatar from '~shared/ColonyAvatar';
 import ChangeColonyLogo from './partials/ChangeColonyLogo';
 import { useActionSidebarContext } from '~context/ActionSidebarContext';
-import { ActionSidebarRowFieldNameEnum } from '~v5/common/ActionSidebarRow/enums';
 
 const displayName = 'v5.common.ActionsContent.partials.ColonyDetailsFields';
 
 const ColonyDetailsFields = () => {
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/ColonyDetailsFields/ColonyDetailsFields.tsx]: Done

File: src/components/v5/common/ActionSidebar/partials/DescriptionField/DescriptionField.tsx [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        >
+          <RichText
+            name={fieldName}
+            isDecriptionFieldExpanded={isDecriptionFieldExpanded}
+            toggleOffDecriptionSelect={toggleOffDecriptionSelect}
+            toggleOnDecriptionSelect={toggleOnDecriptionSelect}
+          />
+        </div>
+      )}
+
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/DescriptionField/DescriptionField.tsx]: Done

File: src/components/v5/common/ActionSidebar/partials/DescriptionField/DescriptionField.tsx [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+    fieldState: { error },
+  } = useController({ name: fieldName });
+  const isError = !!error;
+
+  return (
+    <div className="sm:relative w-full">
+      {!isDecriptionFieldExpanded && (
+        <div
+          className={clsx(
+            'flex text-md transition-colors hover:text-blue-400',
```
[the remark this answers is not in the document]
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/DescriptionField/DescriptionField.tsx]: Done

File: src/components/v5/common/ActionSidebar/partials/PopularActions.tsx (hunk 1 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/PopularActions.tsx]: Done

File: src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx (hunk 2 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/TeamsSelect/TeamsSelect.tsx]: Done

File: src/components/v5/common/ActionSidebar/partials/forms/CreateNewTeamForm/CreateNewTeamForm.tsx (hunk 4 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/partials/forms/CreateNewTeamForm/CreateNewTeamForm.tsx]: Done

File: src/components/v5/common/ActionSidebar/types.ts (hunk 5 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/types.ts:14]: Done

File: src/components/v5/common/SubNavigation/partials/DropdownContent/DecideDropdown.tsx (hunk 6 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/SubNavigation/partials/DropdownContent/DecideDropdown.tsx]: Done

**Review by joanna-pagepro — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx [hunk 20]
```diff
@@ -18,19 +18,103 @@ import PopularActions from './partials/PopularActions';
...
+          </button>
+        )}
+        {children}
+      </div>
+      <div className="px-6 py-8 flex-grow">
+        <input
+          type="text"
+          className={`
+            heading-3 placeholder:text-gray-500
+            hover:text-blue-400 hover:placeholder:text-blue-400 text-gray-900
```
Comment (CzarekDryl) [inline src/components/v5/common/ActionSidebar/ActionSidebar.tsx:85]: Please add hovers only on desktop - Ik it's my code 💀

**Review by CzarekDryl — APPROVED** [review]

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx (hunk 20 above)
Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionSidebar/ActionSidebar.tsx]: `md:` added

**Pull request merged**